storage/bbolt.go

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (C) 2022 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package storage
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/nuts-foundation/go-stoabs"
+	"github.com/nuts-foundation/go-stoabs/bbolt"
+	"github.com/nuts-foundation/nuts-node/storage/log"
+	bboltLib "go.etcd.io/bbolt"
+	"os"
+	"path"
+	"sync"
+	"time"
+)
+
+const fileMode = 0640
+
+type bboltDatabase struct {
+	datadir         string
+	config          BBoltConfig
+	ctx             context.Context
+	cancel          context.CancelFunc
+	shutdownWatcher *sync.WaitGroup
+}
+
+// BBoltConfig specifies config for BBolt databases.
+type BBoltConfig struct {
+	// Backup specifies backup config for the database.
+	Backup BBoltBackupConfig `koanf:"backup"`
+}
+
+// BBoltBackupConfig specifies config for BBolt database backups.
+type BBoltBackupConfig struct {
+	// Directory specifies the directory in which the BBolt backup should be written.
+	Directory string `koanf:"directory"`
+	// Interval specifies the time between backups.
+	Interval time.Duration `koanf:"interval"`
```

Comment:
How is koanf suppose to coerce "1h" to a time.Duration?

Reply:
At first I thought that too and implemented a custom type, but appears it is supported out of the box...